docs/migration-guides/v4.md

```diff
@@ -0,0 +1,398 @@
+> **Why?** In previous versions of Tool Kit, options were plugin-wide. In Tool Kit v4, we have task-specific options, and these can be overriden depending on which command is running a task. This means instead of needing to have e.g. separate `JestLocal` and `JestCi` tasks, we can have a single `Jest` task, with a boolean `ci` option that can be set to `true` for the `test:ci` command only.
+
+Run `npx dotcom-tool-kit --help` again. If you have any commands in your `.toolkitrc.yml` that have been renamed, you'll get a message like:
+
+> These tasks don't exist, but are configured to run from commands:
+>
+> - `JestLocal` (assigned to `test:local` by your app)
+> - `JestCi` (assigned to `test:ci` by your app)
+>
+> They could be misspelt, or defined by a Tool Kit plugin that isn't used by this app.
+
+In your `.toolkitrc.yml`, **rename the tasks to their v4 equivalents**, and specify the appropriate options so you get equivalent behaviour to the old task:
+
+```diff
+ commands:
+   test:local:
+-     - JestLocal
++     - Jest
+   test:ci:
+-     - JestCi
++     - Jest:
++         ci: true
+```
+
+<details><summary><strong>Full list of renamed tasks</strong></summary>
+
+This is a list of all tasks that have been renamed, their v4 equivalents, and the options you need to specify for the same behaviour:
+
+| Renamed task | Tool Kit v4 equivalent | Options to set |
+|-|-|-|
+| `BabelDevelopment` | `Babel` | `envName: 'development'` |
+| `BabelProduction` | `Babel` | `envName: 'production'` |
+| `CypressCi` | `Cypress` | `{}` |
+| `CypressLocal` | `Cypress` | `{}` |
+| `JestCi` | `Jest` | `ci: true` |
+| `JestLocal` | `Jest` | `{}` |
+| `TypeScriptBuild` | `TypeScript` | `{}` |
+| `TypeScriptTest` | `TypeScript` | `noEmit: true` |
+| `TypeScriptWatch` | `TypeScript` | `watch: true` |
+| `WebpackDevelopment` | `Webpack` | `envName: 'development'` |
+| `WebpackProduction` | `Webpack` | `envName: 'production'` |
+| `WebpackWatch` | `Webpack` | `watch: true, envName: 'development'` |
+
+</details>
+
+## 2. Migrate any custom plugins
+
+> [!TIP]
+> If you only use built-in Tool Kit plugins, **you can [skip to step 3](#3-regenerate-config-files-and-test)**.
+>
+> Built-in plugins are listed in your `.toolkitrc.yml` with npm package names, like `@dotcom-tool-kit/webpack`; custom plugins are referenced by relative file path, like `./tool-kit/canary`.
+
+<details><summary><strong>Yes, I have custom plugins</strong></summary>
+
+### 2.1. Steps for all plugins
+
+A plugin's `.toolkitrc.yml` acts as a manifest to tell Tool Kit what to load. To prevent issues when attempting to load an old plugin, there is now a **manifest version**. A `.toolkitrc.yml` without a version is implicitly version 1. Tool Kit v4 will **only load plugins with a version 2 manifest**.
+
+Add the `version` to your plugin's `.toolkitrc.yml` (most plugins' `.toolkitrc.yml` files will be empty before this):
+
+```diff
++version: 2
+```
+
+### 2.2. Plugins with custom tasks
+
+> [!TIP]
```

Comment:
**todo**: this tip isn't rendering properly because it's quoted

Reply:
wdym "because it's quoted"? the blockquote syntax is how you do the tip blocks (i mean, yeah, this one isn't working, but idk why, maybe the code block? it looked fine in vs code preview 😕)

Reply:
ah sorry i just assumed that was why but you know more than me!

Reply:
lmao it's because it's nested in a `<details>` every other bit of markdown syntax is working fine 🥴